### ISSUE TITLE
Dialog tree - select option by number not working as expected

### ISSUE THREAD

**Issue by PascalLadalle** [4ian/GDevelop#1642]
Using the Dialog tree example project, I added more options to the text for testing and set up the key 3 to select option number 3 and confirm the choice. 

Instead of that, hitting key 3 acts like "select previous option" (and doesn't confirm).

![image](https://user-images.githubusercontent.com/25978830/79052933-0dd17680-7c3a-11ea-8787-3651c6abf700.png)
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/25978830/79053063-b4b61280-7c3a-11ea-9d93-a1e432bab600.gif)

**Comment by blurymind** [comment]
can you share the example with the changes you made to it.
I read it twice and I am still confused what the intent is. You want key 3 to select option 3?

**Comment by blurymind** [comment]
I think there are two things going on here:
- you are not using it as its designed to be used
- Gdevelop's event sheet is not properly warning you about the empty parameter field you neglected

**Comment by PascalLadalle** [comment]
> your event sheet logic seems a bit odd. You have select option by number, but you are not giving it a number?
you are not using it as its designed to be used, somehow thinking that your number key will give it the number magically?

No, I'm not expecting magical behavior... Yes, I'm giving it a number, it's just **not displayed** in the events sheet. I guess I should have included this screenshot too:
![image](https://user-images.githubusercontent.com/25978830/79066627-b541be00-7cb9-11ea-8200-a77029ecac31.png)

> You want key 3 to select option 3 and also confirm selection?

Yes, I want key 3 to select the third or fourth option (depending on whether it starts at 0 or 1) and confirm the choice. It's just for **testing**, because I don't understand how this action works, and it's not used in the example. 
The real goal is mouse navigation [for this project](https://gruk.itch.io/gdsolver). I got everything working except for this selection thing.

I read the wiki... But it doesn't say much.

> If you have entered an **actual number value** (not an expression as what most people would do)

I don't understand what you mean. I see a "123" blue button on the side, so it's expecting either a standard number (_hardcoded_, without quotes), either a number variable, right?

> it makes no sense to put a number in that field

Why do you say that? 😞 The action is supposed to "select by number", so why wouldn't I be able to enter a number?  

I'm attaching the project.
[Dialogue tree.zip](https://github.com/4ian/GDevelop/files/4466160/Dialogue.tree.zip)

**Comment by blurymind** [comment]
so what you will need there is to render actual buttons that the user can click on. Unfortunately I haven't tried to do that myself and there is no example out there. 
Without trying it is hard to say if there is a bug or not, but unfortunately don't have time to add that to the dialogue example or create an example for you atm.

Maybe someone might be interested to give this a try? :)

**Comment by PascalLadalle** [comment]
Don't worry about the clicking interface, it is ready, I just need to know **how** this action works, if it works...
How could it accept a number expression and not a number? 🤔  It's very strange! Anyway, I tried: 
![image](https://user-images.githubusercontent.com/25978830/79071596-e089d480-7cdc-11ea-860a-0d1de70cb401.png)
Not working any better. Then I tried:
![image](https://user-images.githubusercontent.com/25978830/79071658-39596d00-7cdd-11ea-99a8-1d00e3ac7cba.png)
![image](https://user-images.githubusercontent.com/25978830/79071840-18454c00-7cde-11ea-82da-7d1c8d5e1af8.png)
and still the same result, it's acting like "select previous option".
Even a value of 0 or a "ToString(3)" does the same thing.

So, it looks like a bug to me...
